### PR TITLE
add a complex example using multiple regularly used plugins related t…

### PR DESCRIPTION
**Pull request by sagiavinash**
add a complex example using multiple regularly used plugins related to the following use-cases

1. build optimization plugins
2. compile time plugins
3. webpack-dev-server enhancement plugins

1. [Read and sign the CLA](https://cla.js.foundation/webpack/webpack.js.org). This needs to be done only once. PRs that haven't signed it won't be accepted.
2. Make sure your PR complies with [the writer's guide](https://webpack.js.org/writers-guide/).
3. Read through the PR diff carefully as sometimes this can reveal issues. The work will be reviewed, but this can save some effort.
4. Remove these instructions from your PR as they are for your eyes only.

### CODE REVIEW

File: content/configuration/plugins.md [hunk 1]
```diff
@@ -23,5 +23,18 @@ plugins: [
 ```
 
 A more complex example, using multiple plugins, might look something like this:
-
+```js
+plugins: [
```
Comment (simon04) [inline content/configuration/plugins.md]: Add the relevant imports:
```
var ExtractTextPlugin = require('extract-text-webpack-plugin');
var DashboardPlugin = require('webpack-dashboard/plugin');
```

File: content/configuration/plugins.md [hunk 2]
```diff
@@ -23,5 +23,18 @@ plugins: [
...
+  // build optimization plugins
+  new webpack.optimize.CommonsChunkPlugin({ name: 'vendor', filename: 'vendor-[hash].min.js' }),
```
Comment (simon04) [inline content/configuration/plugins.md]: Please format all plugin options as follows (this style is used througout the documentation): 
```javascript
new webpack.optimize.CommonsChunkPlugin({
  name: 'vendor',
  filename: 'vendor-[hash].min.js'
}),
```

File: content/configuration/plugins.md [hunk 3]
```diff
@@ -23,5 +23,18 @@ plugins: [
...
+  new ExtractTextPlugin({ filename: 'build.min.css', allChunks: true }),
+  new webpack.IgnorePlugin(/^\.\/locale$/, [/moment$/]),
+  // compile time plugins
+  new webpack.DefinePlugin({ 'process.env.NODE_ENV': '"production"' }),
+  // webpack-dev-server enhancement plugins
+  new DashboardPlugin(),
+  new webpack.HotModuleReplacementPlugin(),
+]
+```      
 ?> Add a more detailed example
```
Comment (simon04) [inline content/configuration/plugins.md]: Remove this TODO since that's what this PR is about.

**Comment by sagiavinash** [comment]
@simon04 i have signed the agreement but its not getting reflected here
Its the first time i am contributing to a js.org repo. am i doing something wrong
here is the screenshot for the signed agreement http://prnt.sc/e56rxe

File: content/configuration/plugins.md (hunk 3 above)
Comment (sagiavinash) [inline content/configuration/plugins.md]: got it

**Comment by sagiavinash** [comment]
@simon04 thanks for reviewing. I have updated my PR

File: content/configuration/plugins.md [hunk 4]
```diff
@@ -23,5 +23,36 @@ plugins: [
 ```
 
 A more complex example, using multiple plugins, might look something like this:
+```js
+// importing plugins that dont come by default in webpack
```
Comment (simon04) [inline content/configuration/plugins.md]: s/dont/do not/

Comment (sagiavinash) [inline content/configuration/plugins.md]: got it

**Comment by simon04** [comment]
You committed using an email address not associated with your GitHub user. See the second part of https://github.com/webpack/webpack.js.org/pull/834#issuecomment-277667274

**Comment by sagiavinash** [comment]
@simon04
- fixed the commits user meta to comply with CLA agreement.
- CI tests also passed

Can your merge it.

**Pull request merged**